vue/src/views/LokiLogs.vue

```diff
@@ -126,39 +129,44 @@ export default defineComponent({
     })
 
     function onClickFilter(filter: Filter) {
-      query.value = updateQuery(query.value, filter.key, filter.op, JSON.stringify(filter.value))
+      query.value = queryParser(query.value, filter.key, filter.value, filter.selected, filter.op)
     }
 
     return { query, limit, logql, onClickFilter }
   },
 })
 
-const STREAM_SEL_RE = /{[^}]*}/
-
-function updateQuery(query: string, key: string, op: string, value: string): string {
-  const selector = `${key}${op}${value}`
-
-  if (!query) {
-    return `{${selector}}`
-  }
-
-  const m = query.match(STREAM_SEL_RE)
-  if (!m) {
-    return `{${selector}}`
-  }
-
-  let found = m[0]
-
-  const e = escapeRe
-  const re = new RegExp(`${e(key)}\\s*${e(op)}\\s*("(?:[^"\\\\]|\\\\.)*")`)
-  if (re.test(found)) {
-    found = found.replace(re, selector)
+// stream selector regex
+const STREAM_SELECTOR_REGEX = /{[^}]*}/
+
+export function queryParser(
```

Comment:
Could you work a bit on the names? We have `queryParse` and `queryParser` which is rather confusing.

Also, perhaps move this function to decodeQuery.ts?